### DISCUSSION TITLE
Print Page Issue In Pimcore (Web-to-Print)

### DISCUSSION THREAD

**Discussion by KasthuriAnandhan** [pimcore/pimcore#15568] [Q&A / Support]
@blankse

Hi Sebastian Blank, there is no print pages in documents of the pimcore, except this everything will be showed how to add print pages I have installed web-to-print bundle also but can't to view the print pages in documents of the pimcore, can you please help me to resolve this issue and kindly provide the Web-to-print-settings configuration which is under the settings of the pimcore. Below I have attached the screen shots of my enterprise pimcore and demo version documents.

My Enterprise Pimcore's Document
![Demo Screenshot ](https://user-images.githubusercontent.com/125338676/254531817-db978a9c-7382-454b-ac15-f58d62a14706.png)

Demo version Pimcore's Document
![Demo Screenshot ](https://github.com/pimcore/pimcore/assets/125338676/5b64d6b7-5cff-4f45-869a-bf8e3739fc06)

Can you see the difference between these two, kindly give the solution to enable print pages.

Thank You!

**Comment by KasthuriAnandhan** [comment]
@kingjia90

Hi JiaJia Ji kindly help me to resolve my issue.

**Reply by kingjia90** [reply]
Excluded that the bundle is not activated by checkbox and confirmed that there are no js console error.

Please check if the user has perspective permission set
https://github.com/pimcore/pimcore/blob/48e8b133c3ffdf091e36cc9abd9cb335bb8d44f0/bundles/AdminBundle/Resources/public/js/pimcore/document/tree.js#L400

Other than that, don't know any other things that could stop from displaying

**Comment by KasthuriAnandhan** [comment]
Yeah the user have permission for perspective set and also the user is an admin user.

![Screenshot 2023-07-20 143033](https://github.com/pimcore/pimcore/assets/125338676/e40d5a63-8b78-446e-bc52-c7545e65c0dc)

**Comment by KasthuriAnandhan** [comment]
Should I create any perspective and custom views to enable print pages.

**Comment by NiklasBr** [comment]
Can you run this CLI command and show the result here?

`bin/console pimcore:bundle:list | grep -E "Print|Installed"`

**Reply by kingjia90** [reply]
Thank you for joining 😄 
It seems that it's on 10.x 
See also https://github.com/pimcore/web-to-print-bundle/issues/42#issuecomment-1643476378

**Comment by KasthuriAnandhan** [comment]
![Screenshot 2023-07-20 144006](https://github.com/pimcore/pimcore/assets/125338676/77546b4e-7cf4-4438-8f6d-61769b88506c)

**Comment by KasthuriAnandhan** [comment]
Hi @NiklasBr & @kingjia90 

Can't to resolve my issue?☹️

**Comment by KasthuriAnandhan** [comment]
Hello @NiklasBr & @kingjia90

Can you please help me here to resolve my issue!!

**Reply by kingjia90** [reply]
Suggesting to try again but with Direct Edit bundle disabled
https://github.com/pimcore/web-to-print-bundle/issues/42#issuecomment-1643485885

If it's not working then, please contact enterprise support, probably it would end up having a remote view/control to check for this specific case.  Code-wise there are no breaking bugs and i am not noticing anything misconfigured or wrongly installed so far.

**Comment by KasthuriAnandhan** [comment]
Tried as you said but no difference anyways thanks for your time and response, we are contacting the enterprise support team to resolve this issue.

Thank You Again😊 @kingjia90

**Comment by kingjia90** [comment]
@KasthuriAnandhan Thank you for the patience, we are looking into this again
 
There are further things to try/check, could you disable Headless Documents to see if it there's a conflict on which sub-item should appear on top of context menu?
If possible, could you also share the version of Headless Document installed, just to rule out that the problem is related to https://github.com/pimcore/headless-documents/pull/128/

Also can you please try to add a custom document type that has "print page/container" as base type?
![Screenshot from 2023-08-23 11-03-05](https://github.com/pimcore/pimcore/assets/6014195/18b1af1f-c4a9-4dff-8050-abeb4e65b9cb)
just to see if these document types are not registered or is just a visual stuff caused by headless documents

**Comment by KasthuriAnandhan** [comment]
Hi  @kingjia90 sorry for the late response below I have shared the version of headless document.
![image](https://github.com/pimcore/pimcore/assets/125338676/0c1b4834-6294-468e-8dbc-86e0ea8e6f6a)

**Comment by KasthuriAnandhan** [comment]
Tried as you said then also can't be able to see Add PrintPage.

![Screenshot 2023-08-24 173220](https://github.com/pimcore/pimcore/assets/125338676/db84b826-3f44-482a-a8b4-d80454d1cb3a)

![Screenshot (235)](https://github.com/pimcore/pimcore/assets/125338676/1a7a60d7-f984-4823-8e25-2e094deb872a)

**Comment by kingjia90** [comment]
@KasthuriAnandhan   We are trying to reproduce but it seem all working fine, there's only one possible case that it breaks like that and that the cache is not cleared. Can you please try to clear all the cache and check again?

**Reply by dvesh3** [reply]
Please also check if there are any console errors?

**Reply by KasthuriAnandhan** [reply]
Okay will try and let you know sorry for the late response

**Comment by KasthuriAnandhan** [comment]
Hi @kingjia90 @dvesh3,

I have installed the Web2Print bundle on my local Pimcore instance. I can now see the PrintPage and PrintContainer under Documents, and I’ve managed to generate a DataObject in PDF format, similar to what is available in the Pimcore demo version. However, I’m not sure where or how to start to create a PDF with DataObject information.

Could you please assist me with this? I've attached some screenshots for your reference.

![Screenshot (335)](https://github.com/user-attachments/assets/121cc0f0-57d2-43c1-bb08-141471cea189)
![Screenshot (334)](https://github.com/user-attachments/assets/3ef28966-92a9-439c-9f77-fdc136c1504d)
[composer.json](https://github.com/user-attachments/files/16580427/composer.json)

Thanks,
Kasthuri Anandhan

**Reply by kingjia90** [reply]
What do you mean specifically? 
Just do as usual for any document (twig, editables, controller/template, properties etc..), the only difference is that instead of generating an html, it would convert it into a pdf.

https://github.com/pimcore/web-to-print-bundle/blob/1.x/doc/img/printdocuments_flow.png

**Comment by KasthuriAnandhan** [comment]
Hi @kingjia90 ,

I'm just testing to see if I can create a PDF. I added a simple print statement in the PrintPage and tried to generate the PDF, but it hasn't completed. The progress has been stuck at 0% for a long time.

![Screenshot 2024-08-13 102043](https://github.com/user-attachments/assets/bade95eb-27df-4e86-9dfd-ff80d057c3f6)
![Screenshot 2024-08-13 102109](https://github.com/user-attachments/assets/a7e1deb7-29ec-4368-b182-ab62b7a9ffdf)

And the below bricks are not available for me how to enable those also.
![Screenshot 2024-08-13 101916](https://github.com/user-attachments/assets/04fecb13-c3de-4bd5-b315-efdb248b7af7)

**Reply by kingjia90** [reply]
The progress 0% could be due to maintenance cronjobs not being set, see also https://github.com/pimcore/pimcore/discussions/15364#discussioncomment-6172175

For the editable issue, please edit the twig template as used to or create a new template with the editable you want to provide to the users

See also
https://github.com/pimcore/demo/blob/2024.x/templates/web2print/default_no_layout.html.twig

**Comment by KasthuriAnandhan** [comment]
Hi @kingjia90 ,

Sorry to disturb you, but I have a doubt. Can I follow the same steps on my local Pimcore installation as well for maintenance cronjobs ?

**Reply by kingjia90** [reply]
Yes, if i got the question right, locally one could activate all and there are no much worries.
Some tweaking may be required when going to production, as mentioned [here](https://pimcore.com/docs/platform/Pimcore/Getting_Started/Installation/Webserver_Installation/#:~:text=we%20recommend%20following%20the%20symfony%20messenger%20guide%20for%20setting%20up%20worker%20processes%20in%20production.) but that depends on the project real needs

**Comment by KasthuriAnandhan** [comment]
Hi @kingjia90,

I’m unsure how to manage cron jobs in my local Pimcore environment. Could you provide more information on this? I’ve already set up a task scheduler, but I’m still encountering issues. I’m trying to create a PDF from PrintPage using a simple print statement. Could you please help me with this?

Thank you!

**Reply by kingjia90** [reply]
doing as described https://pimcore.com/docs/platform/Pimcore/Getting_Started/Installation/Webserver_Installation/#5-maintenance-cron-job should be fine, i have it taken care by supervisord https://github.com/pimcore/skeleton/blob/2024.x/.docker/supervisord.conf#L5 locally as i am using docker

**Comment by KasthuriAnandhan** [comment]
Hi @kingjia90, 

I'm unable to resolve this issue in my local Pimcore environment. If I install the bundle from the link below, do you think it will help me overcome the 0% issue?

https://github.com/samynw/pimcore-maintenance-toolbox

**Comment by KasthuriAnandhan** [comment]
Hi @dvesh3,

I've noticed a few similar issues on GitHub regarding PDF generation. I've installed Pimcore on my local Windows machine and am trying to generate a PDF using PrintPage and PrintContainer, but the process remains stuck at 0% for a long time. If you have any insights or suggestions on how to fix this, I would greatly appreciate your help.
